### PR TITLE
Add ability to define conditionals for counter_cache

### PR DESCRIPTION
**Pull request by Gargron**
E.g.:

```ruby
counter_cache :statuses,
              if: ->(status) { status.public_visibility? },
              scope: ->(statuses) { statuses.where(visibility: :public) }
```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -15,7 +15,10 @@ gem 'counter_cache-rails'
 ```rb
 class Post
   has_many :comments
-  counter_cache :comments
+
+  counter_cache :comments,
+                if: ->(comment) { comment.visible? },
```
Comment (iguchi1124) [inline README.md:20]: I think rails programmer expects `:unless` option is available when `:if` option was available, but, I think this option's mainly use case is that judgment target record is countable whether or not.

So, I guess that option should be named to "countable",  or, should be support "unless" option. what do you think?

**Review by iguchi1124 — COMMENTED** [review]
@Gargron thanks for great proposal!

**Comment by Gargron** [comment]
@iguchi1124 Addressed. By the way, what if Rails configuration has a global expiration option? These counters would be removed after 10 minutes in Mastodon, I think...

**Comment by iguchi1124** [comment]
Thanks for good feedback! I missed about global cache expiration option until now. I will work to support expiration configuration for this gem.

**Pull request merged**